### DISCUSSION TITLE
Is it plan to released on one day in the future?

### DISCUSSION THREAD

**Discussion by Myoko** [garyttierney/me3#12] [General]
does this tool have good compatibility with ER's DLC? This looks great! I hope to see a pre-release version soon.

**Comment by garyttierney** [comment]
Yes. you can consider this thread to be a tracker on the release.

**Comment by Myoko** [comment]
awesome, I will keep tracker on the release.

**Comment by AlDollaz** [comment]
Nice!

**Comment by nex3** [comment]
Definitely interested in switching the DS3 [Archipelago](https://archipelago.gg) randomizer to this once it's stable.

**Comment by bdzerr** [comment]
Hey! Can we expect a release of some sort anytime soon?

**Comment by garyttierney** [comment]
It's now available for ELDEN RING and support for other titles from DS3 through to NIGHTREIGN should be available soon.

**Comment by nex3** [comment]
Fantastic news. Is there a guide somewhere for writing mods that target ME3, or an example we can follow?

**Comment by garyttierney** [comment]
There's a lot of gaps in the docs right now for mod authors, will be polishing that tonight after cutting v0.4.0 for NR/AC6 support. For now there's this quick reference: https://me3.readthedocs.io/en/latest/user-guide/creating-mod-profiles.html.

What isn't mentioned there that's fairly important is being able to use a `supports` key to list the games your profile supports (see a profile created with `me3 profile create -g er`). That gets you double-click to launch in the Windows shell or on desktop environments supporting FreeDesktop desktop entry files.

**Comment by nex3** [comment]
Great! Is there an equivalent to `modengine_ext_init` and the C++ inheritance structure that ME2 used, or are mods just supposed to start doing whatever from `DllMain`?

**Comment by garyttierney** [comment]
Not right now, the most you can do is provide a symbol that me3 will call until it returns true as a signal your mod is ready (see NativeInitializerCondition in the profile schema). The goal is to give mod authors direct access to the APIs by compiling against the same toolchain as us, but that isn't quite ironed out yet.

**Reply by nex3** [reply]
Got it. Is there an expectation that those mods will be written in Rust?

**Reply by garyttierney** [reply]
Initially yeah, we'll also provide a simpler C API that has looser ABI restrictions but I think exposing all the core Rust code via a public rust-call ABI will be the easiest way to get something working.

**Reply by nex3** [reply]
That'll require a rewrite of our Archipelago mod, but honestly that's overdue anyway.

**Reply by garyttierney** [reply]
Re-looking at this I think it's trivial for us to provide a header-only library that shims the ModEngine2 extension API over a me3 C API since it's not doing anything fancy like JITing landing pads for closures. The only thing that'd require more than a simple shim would be the `create_config_reader` stuff.
